daemon/image_delete.go

```diff
@@ -148,21 +156,55 @@ func (daemon *Daemon) ImageDelete(imageRef string, force, prune bool) ([]types.I
 				return nil, conflict
 			}
 
-			parsedRef, err := daemon.removeImageRef(repoRefs[0])
-			if err != nil {
-				return nil, err
-			}
+			for _, repoRef := range repoRefs {
+				parsedRef, err := daemon.removeImageRef(repoRef)
+				if err != nil {
+					return nil, err
+				}
 
-			untaggedRecord := types.ImageDelete{Untagged: parsedRef.String()}
+				untaggedRecord := types.ImageDelete{Untagged: parsedRef.String()}
 
-			daemon.LogImageEvent(imgID.String(), imgID.String(), "untag")
-			records = append(records, untaggedRecord)
+				daemon.LogImageEvent(imgID.String(), imgID.String(), "untag")
+				records = append(records, untaggedRecord)
+			}
 		}
 	}
 
 	return records, daemon.imageDeleteHelper(imgID, &records, force, prune, removedRepositoryRef)
 }
 
+// isSingleReference tests a list of references to see if either
+// references from multiple repositories are represented or if
+// there is only a single tag reference. Canonical references to
+// the same repository as a tag are ignored.
+func isSingleReference(repoRefs []reference.Named) bool {
+	if len(repoRefs) <= 1 {
+		return len(repoRefs) == 1
+	}
+	var singleRef reference.Named
+	canonicalRefs := []reference.Canonical{}
+	for _, repoRef := range repoRefs {
+		if cr, isCanonical := repoRef.(reference.Canonical); isCanonical {
+			canonicalRefs = append(canonicalRefs, cr)
+		} else if singleRef == nil {
+			singleRef = repoRef
+		} else {
+			return false
+		}
+	}
+	if singleRef == nil {
+		// Just use first canonical ref
+		singleRef = repoRefs[0]
+	}
+	for _, canonical := range canonicalRefs {
+		if canonical.Name() != singleRef.Name() {
+			return false
+		}
+	}
```

Comment:
I think that is better, I will also need to test the length, `return ok && len(canonicalRefs) == 1` since this is testing there is not entries from other repositories as well.  Going to update the comment to make that clearer